File: 1503.所有蚂蚁掉下来前的最后一刻.py
#
# @lc app=leetcode.cn id=1503 lang=python3
#
# [1503] 所有蚂蚁掉下来前的最后一刻
#

# @lc code=start
from typing import List
class Solution:
    def getLastMoment(self, n: int, left: List[int], right: List[int]) -> int:
        return max(max(left, default=0), n - min(right, default=n))
# @lc code=end

File: 1503.所有蚂蚁掉下来前的最后一刻.cpp
/*
 * @lc app=leetcode.cn id=1503 lang=cpp
 *
 * [1503] 所有蚂蚁掉下来前的最后一刻
 */

// @lc code=start
#include<vector>
using namespace std;
class Solution {
public:
    int getLastMoment(int n, vector<int>& left, vector<int>& right) {
        return max((right.empty()? 0 : n - *min_element(right.begin(), right.end())), (left.empty()? 0 : *max_element(left.begin(), left.end())));
    }
};
// @lc code=end

